src/clj_sqs_extended/working_example.clj

```diff
@@ -0,0 +1,142 @@
+(ns clj-sqs-extended.working-example
+  "Clojure version of the AWS working example found at:
+   https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-s3-messages.html"
+  (:require [clojure.tools.logging :as log]
+            [tick.alpha.api :as t])
+  (:import [java.util UUID]
+           [com.amazonaws.client.builder AwsClientBuilder$EndpointConfiguration]
+           [com.amazonaws.services.s3 AmazonS3ClientBuilder]
+           [com.amazonaws.services.s3.model
+              BucketLifecycleConfiguration
+              BucketLifecycleConfiguration$Rule
+              ListVersionsRequest]
+           [com.amazonaws.services.sqs AmazonSQSClientBuilder]
+           [com.amazonaws.services.sqs.model
+              CreateQueueRequest
+              SendMessageRequest
+              DeleteMessageRequest
+              ReceiveMessageRequest]
+           [com.amazon.sqs.javamessaging
+              AmazonSQSExtendedClient
+              ExtendedClientConfiguration]))
+
+
+(defn- random-bucket-name
+  "Creates a random name for the test bucket."
```

Comment:
this docstring seems redundant as the fn name is self-explanatory enough. it's a balance between clarity versus verbosity so just FYI in this case. everyone has their preference.